common/socket/src/main/java/io/helidon/common/socket/IdleInputStream.java

```diff
@@ -36,14 +40,25 @@
  */
 class IdleInputStream extends InputStream {
 
+    /**
+     * This needs to cooperate with 100-continue timeout.
+     * Some low number is needed here for a quick iterations.
+     */
+    private static final int TIMEOUT = 100;
```

Comment:
Should it be configurable ?

Reply:
I think it does not necessarily need to be. Or do you?

Reply:
The value was 500, it's now 100 and it's a value we picked somewhat arbitrarily.
So why not make it configureable from the start.

Reply:
The 100 was chosen on my request. We need to have a reasonable small time, yet something that will not burn too much CPU.
The default timeout for 100 continue is 1 second, so we expect to spend max. 1.1 second waiting for it. Each higher number means more "error" introduced.

Of course if somebody sets a really small timeout for 100 continue, they would get up to 100 millis added on the first request (further request will skip 100 continue). I think this is a reasonable behavior for now.

Also configuration is not available in sockets, and this would cause a ripple effect in all APIs that use them. If we need to add configurability, we  would also need to add an object to send it around.